packages/shared/src/hooks/translation/useTranslation.ts

```diff
@@ -0,0 +1,141 @@
+  queryType: 'post' | 'feed';
+}) => {
+  fetchTranslations: (id: string[]) => void;
+};
+
+type TranslateEvent = {
+  id: string;
+  title: string;
+};
+
+const updateTranslation = (post: Post, translation: TranslateEvent): Post => {
+  const updatedPost = post;
+  if (post.title) {
+    updatedPost.title = translation.title;
+    updatedPost.translation = { title: !!translation.title };
+  } else {
+    updatedPost.sharedPost.title = translation.title;
+    updatedPost.sharedPost.translation = { title: !!translation.title };
+  }
+
+  return updatedPost;
+};
+
+export const useTranslation: UseTranslation = ({ queryKey, queryType }) => {
+  const abort = useRef<AbortController>();
+  const { user, accessToken, isLoggedIn } = useAuthContext();
+  const queryClient = useQueryClient();
+
+  const { language } = user || {};
+
+  const updateFeed = useCallback(
+    (translatedPost: TranslateEvent) => {
+      const updatePost = updateCachedPagePost(queryKey, queryClient);
+      const feedData =
+        queryClient.getQueryData<InfiniteData<FeedData>>(queryKey);
+      const { pageIndex, index } = findIndexOfPostInData(
+        feedData,
+        translatedPost.id,
+        true,
+      );
+      if (index > -1) {
+        updatePost(
+          pageIndex,
+          index,
+          updateTranslation(
+            feedData.pages[pageIndex].page.edges[index].node,
+            translatedPost,
+          ),
+        );
+      }
+    },
+    [queryKey, queryClient],
+  );
+
+  const updatePost = useCallback(
+    (translatedPost: TranslateEvent) => {
+      updatePostCache(queryClient, translatedPost.id, (post) =>
+        updateTranslation(post, translatedPost),
+      );
+    },
+    [queryClient],
+  );
+
+  const fetchTranslations = useCallback(
+    async (postIds: string[]) => {
+      if (!isLoggedIn || !language) {
+        return;
+      }
+      if (postIds.length === 0) {
+        return;
+      }
+
+      const params = new URLSearchParams();
+      postIds.forEach((id) => {
+        params.append('id', id);
+      });
+
+      const messages = await stream(
+        `${apiUrl}/translate/post/title?${params}`,
+        {
+          headers: {
+            Authorization: `Bearer ${accessToken?.token}`,
+            'Content-Language': language as string,
+          },
+        },
+      );
+
+      // eslint-disable-next-line no-restricted-syntax
+      for await (const message of messages) {
+        if (message.event === ServerEvents.Message) {
+          const post = JSON.parse(message.data) as TranslateEvent;
+          if (queryType === 'feed') {
+            updateFeed(post);
+          } else {
+            updatePost(post);
+          }
+        }
+      }
+    },
+    [
+      accessToken?.token,
+      isLoggedIn,
+      language,
+      queryType,
+      updateFeed,
+      updatePost,
+    ],
+  );
+
+  useEffect(() => {
+    abort.current = new AbortController();
+
+    return () => {
+      abort.current?.abort();
+    };
```

Comment:
How does this abort `stream`, don't see how its connected to it?

Reply:
Might've forgotten to add the abort controller 😬

Reply:
Added it now